src/Avalonia.Controls.DataGrid/Collections/DataGridSortDescription.cs

```diff
@@ -170,8 +170,29 @@ private IComparer GetComparerForType(Type type)
                 if (type == typeof(string))
                     return _cultureSensitiveComparer.Value;
                 else
+                    return GetComparerForNotStringType(type);
+            }
+
+            internal static IComparer GetComparerForNotStringType(Type type)
+            {
+#if NET6_0_OR_GREATER
+                if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(Nullable<>) && type.GetGenericArguments()[0].IsAssignableTo(typeof(IComparable)))
+                    return Comparer<object>.Create((x, y) => 
+                    {
+                        if (x == null)
+                            return y == null ? 0 : -1;
+                        else
+                            return (x as IComparable)!.CompareTo(y);
+                    });
+                else if (type.IsAssignableTo(typeof(IComparable)))
+                    return Comparer<object>.Create((x, y) => (x as IComparable)!.CompareTo(y));
+                else
+                    return Comparer<object>.Create((x, y) => 0); //avoid using reflection to avoid crash on AOT
```

Comment:
On NET6_0_OR_GREATER you can check for this property whether reflection is supported: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimefeature.isdynamiccodesupported?view=net-8.0

On AOT builds IsDynamicCodeSupported returns false, and this fallback would make sense there.

Reply:
This way we also won't regress any application that realizes on this reflection sorting right now.

Reply:
Nice suggestion. Done.